workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityComponentConversionGenerator.tt

```diff
@@ -513,16 +514,25 @@ namespace <#= qualifiedNamespace #>
                 ComponentType.ReadOnly<SpatialEntityId>()
             };
 
-            public override ComponentType[] CommandTypes => new ComponentType[] {
 <# foreach (var commandDetails in commandDetailsList) { #>
-                ComponentType.ReadOnly<<#= qualifiedNamespace #>.<#= componentDetails.ComponentName #>.CommandSenders.<#= commandDetails.CommandName #>>(),
-                ComponentType.ReadOnly<<#= qualifiedNamespace #>.<#= componentDetails.ComponentName #>.CommandResponders.<#= commandDetails.CommandName #>>(),
+            private CommandStorages.<#= commandDetails.CommandName#> <#= commandDetails.CamelCaseCommandName #>Storage;
 <# } #>
-            };
 
+            private EntityArchetypeQuery[] CommandQueries =
```

Comment:
Does this need to be an array if we are unwrapping it in the SendCommands function? Is it used somewhere else?

Reply:
if we have multiple commands in a component, you will need a query for every type of command